spec/compiler/crystal/tools/doc_spec.cr

```diff
@@ -0,0 +1,43 @@
+require "../../../spec_helper"
+
+GIT_REMOTE_PATTERNS = Crystal::Doc::Generator::GIT_REMOTE_PATTERNS.keys
+private def assert_matches_pattern(url, **options : Object)
+  match = GIT_REMOTE_PATTERNS.each.compact_map(&.match(url)).first?
+  if match
+    options.each { |k, v| match[k.to_s].should eq(v) }
+  end
+end
+
+describe Crystal::Doc::Generator do
+  describe "GIT_REMOTE_PATTERNS" do
```

Comment:
Definitely an improvement. Another approach would be to refactor `#compute_repository` to allow specs for the potential outputs of `git remote -v`. But I'm ok on keeping it this way for now.